src/saga/disks.js

```diff
@@ -77,25 +77,39 @@ function* editDiskOnVm (action) {
   yield fetchDisks({ vms: [ { id: vmId } ] })
 }
 
-function* waitForDiskToBeUnlocked (vmId, diskId) {
-  return yield waitForDisk(vmId, diskId, disk => disk.status && disk.status !== 'locked')
+function* waitForDiskToBeUnlocked (vmId, attachmentId) {
+  return yield waitForDiskAttachment(
+    vmId,
+    attachmentId,
+    attachment => attachment.disk && attachment.disk.status && attachment.disk.status !== 'locked',
+  )
 }
 
 // TODO: drop polling in favor of events (see https://github.com/oVirt/ovirt-web-ui/pull/390)
-function* waitForDisk (vmId, diskId, test, canBeMissing = false) {
+function* waitForDiskAttachment (vmId, attachmentId, test, canBeMissing = false) {
   let metTest = false
 
   for (let i = 1; !metTest && i < 20; i++) {
-    const apiDisk = yield callExternalAction('disk', Api.disk, { payload: { diskId } }, canBeMissing)
-    const edited = Api.diskToInternal({ disk: apiDisk })
-    if (vmId) {
-      yield put(updateVmDisk({ vmId, disk: edited }))
+    const apiDiskAttachment = yield callExternalAction(
+      'diskattachment',
+      Api.diskattachment,
+      { payload: { vmId, attachmentId } },
+      canBeMissing
+    )
+
+    if (!apiDiskAttachment.error) {
+      const apiDisk = apiDiskAttachment.disk
+      const edited = Api.diskToInternal({ attachment: apiDiskAttachment, disk: apiDisk })
+      if (vmId) {
+        yield put(updateVmDisk({ vmId, disk: edited }))
+      }
     }
 
-    if (test(apiDisk)) {
+    if (test(apiDiskAttachment)) {
       metTest = true
     } else {
-      yield delay(Math.log2(i) * 2000) // gradually wait a bit longer each time
+      const delayMs = Math.log2(i + 1) * 2000
```

Comment:
Is `i + 1` necessary and why `i + 1` is better then `i`?

Reply:
`Math.log2(1) = 0`
`Math.log2(2) = 1`

So instead of starting the for loop at 2, I just did the +1 here.

Reply:
I have a better idea to wrap the `delayMs` generation in a generator function.  Looks slick.